[Package]: nelua

### Package description

Minimal, efficient, statically-typed and meta-programmable systems programming language heavily inspired by Lua, which compiles to C and native code.

### Home page URL

https://nelua.io

### Source code URL

https://github.com/edubart/nelua-lang

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

Everything you need is in the GitHub repository and a simple `make` gets the job done. This should be an easy addition.

https://github.com/2096779623/termux-packages/runs/8050524179?check_suite_focus=true
```shell
gcc \
	-DMAXRECLEVEL=400 -DLUA_USE_RPMALLOC -DNDEBUG -DLUA_USE_LINUX  \
	-Isrc/lua  \
	-std=gnu99 -O2  \
	src/hasher.c src/lfs.c src/luainit.c src/onelua.c src/sys.c src/lpeglabel/lpcap.c src/lpeglabel/lpcode.c src/lpeglabel/lpprint.c src/lpeglabel/lptree.c src/lpeglabel/lpvm.c src/srpmalloc/srpmalloc.c  \
	-o nelua-lua \
	-L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -fopenmp -static-openmp -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,relro,-z,now -Wl,-E  -lm -ldl 
gcc: error: unrecognized command-line option ‘-static-openmp’
make: *** [Makefile:81: nelua-lua] Error 1
Error: Process completed with exit code 2.
```

I can't find `-static-openmp` in Makefile :p .

--- a/packages/nelua/Makefile.patch
+++ b/packages/nelua/Makefile.patch
@@ -1,0 +1,29 @@
+diff -uNr nelua-lang/Makefile nelua-lang.mod/Makefile
+--- nelua-lang/Makefile	2022-08-28 12:14:53.076791242 +0800
++++ nelua-lang.mod/Makefile	2022-08-28 12:16:10.306791213 +0800
+@@ -40,7 +40,6 @@
+ OPT_CFLAGS=-O3 -flto -fno-plt -fno-stack-protector
+ ifeq ($(SYS), Linux)
+ 	CFLAGS=-std=gnu99 -O2
+-	CC=gcc
+ 	DEFS+=-DLUA_USE_LINUX
+ 	LIBS+=-lm -ldl
+ 	LDFLAGS+=-Wl,-E
+@@ -227,7 +226,7 @@
+ # Install
+ 
+ # Install paths
+-PREFIX=/usr/local
++PREFIX=@TERMUX_PREFIX@
+ DPREFIX=$(DESTDIR)$(PREFIX)
+ PREFIX_BIN=$(DPREFIX)/bin
+ PREFIX_LIB=$(DPREFIX)/lib
+@@ -350,7 +349,7 @@
+ 
+ CACHE_DIR=.cache
+ ifdef HOME
+-	CACHE_DIR=$(HOME)/.cache/nelua
++	CACHE_DIR=@TERMUX_HOME@/.cache/nelua
+ else ifdef USERPROFILE
+ 	CACHE_DIR=$(USERPROFILE)\\.cache\\nelua
+ endif

--- a/packages/nelua/build.sh
+++ b/packages/nelua/build.sh
@@ -1,0 +1,10 @@
+TERMUX_PKG_HOMEPAGE=https://nelua.io
+TERMUX_PKG_DESCRIPTION="Minimal, efficient, statically-typed and meta-programmable systems programming language"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=2022.07.21
+TERMUX_PKG_SRCURL=https://github.com/edubart/nelua-lang.git
+TERMUX_PKG_GIT_BRANCH="master"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_DEPENDS="build-essential, gdb, git"
+TERMUX_PKG_SUGGESTS="sdl2"

